utils/llm.py

```diff
@@ -2,130 +2,98 @@
 from typing import Optional, Callable
 
 import openai
-import tiktoken
-from openai import OpenAI
-from utils.ui import colour_print
+import litellm
+from litellm import completion
+
+litellm.modify_params = True  # To allow it adjusting the prompt for Claude LLMs
 
 LLMMessage = dict[str, str]
 LLMContext = list[LLMMessage]
 # This list should only contain exact IDs for latest models.
 # Add previous ID if specifically supported.
-SUPPORTED_MODELS: dict[str, tuple[int, tuple[float, float]]] = {
-    "gpt-3.5-turbo-0125": (16_384, (5e-7, 1.5e-6)),
-    "gpt-4": (8_192, (3e-5, 6e-5)),
-    "gpt-4-32k": (32_768, (6e-5, 1.2e-4)),
-    "gpt-4-1106-preview": (128_000, (1e-5, 3e-5)),
-    "gpt-4-0125-preview": (128_000, (1e-5, 3e-5)),
-    "claude-2.1": (200_000, (8e-6, 2.4e-5)),
-    "claude-3-haiku-20240229": (200_000, (2.5e-7, 1.25e-6)),
-    "claude-3-sonnet-20240229": (200_000, (3e-6, 1.5e-5)),
-    "claude-3-opus-20240229": (200_000, (1.5e-5, 7.5e-5)),
-}
 MODEL_ALIASES = {
     "gpt-3.5-turbo": "gpt-3.5-turbo-0125",
-    "gpt-4-turbo": "gpt-4-0125-preview",
+    "gpt-4-turbo": "gpt-4-turbo-2024-04-09",
     "gpt-4-1106": "gpt-4-1106-preview",
-    "gpt-4-0125": "gpt-4-0125-preview",
     "claude-3-haiku": "claude-3-haiku-20240229",
     "claude-3-sonnet": "claude-3-sonnet-20240229",
     "claude-3-opus": "claude-3-opus-20240229",
 }
 GPT_CHEAPEST = "gpt-3.5-turbo"
-GPT_4_TURBO_BEST = "gpt-4-1106-preview"
-client = OpenAI()
+GPT_4_TURBO_BEST = "gpt-4-turbo"
+CHEAPEST_TOKENISER = "claude-3-opus"
```

Comment:
why cheapest? if it's because it gives a higher token count, I'd use a different name that expresses that.